### PR TITLE
Azure disks module with role

### PR DESCRIPTION
**Pull request by gustavoborragan**

### CODE REVIEW

**Review by jcframil — APPROVED** [review]
lgtm

File: modules/azure-disks/README.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
+# Azure disks with role
+
+## Overview
+
+This module creates azure disks with role into Azure provider.
```
Comment (alambike) [inline modules/azure-disks/README.md:5]:
```suggestion
This module creates azure disks and assign a role to manage them, to a service principal.
```

File: modules/azure-disks/README.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+resource_group_name: "REDACTED-RESOURCE-GROUP"
+location: "REDACTED-LOCATION"
+disks:
+    - name: disk-1
+      storage_account_type: StandardSSD_LRS
+    - name: disk-2
+    - name: disk-3
+    - name: disk-4
+      storage_account_type: Premium_LRS
+principal_id: "XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX"
```
Comment (alambike) [inline modules/azure-disks/README.md:61]: Missing role attribute.

Also role and principal_id should be optional. This module could be useful to create a set of disk, but you could not need to assign a role to any principal to manage them (maybe for example, the principal already has that permission through the resource group)

File: modules/azure-disks/main.tf [hunk 3]
```diff
@@ -0,0 +1,25 @@
+# https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/managed_disk
+resource "azurerm_managed_disk" "disks" {
+  for_each             = { for disk in var.disks : disk.name => disk }
+  name                 = each.key
+  location             = var.location
+  resource_group_name  = var.resource_group_name
+  storage_account_type = lookup(each.value, "storage_account_type", "StandardSSD_LRS")
+  create_option        = lookup(each.value, "create_option", "Empty")
```
Comment (alambike) [inline modules/azure-disks/main.tf:8]: An option to create the disk from an snapshot could be very useful, but I think is already supported in the module, maybe just an usage example should be enough.

**Review by alambike — APPROVED** [review]

**Review by alambike — COMMENTED** [review]
Please, take a look to the previous review comments: https://github.com/prefapp/tfm/pull/302#pullrequestreview-2367664654

**Pull request merged**